src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,225 @@
+
+Typical usage example::
+
+    $ pytest test_orthology_benchmark.py
+
+"""
+
+from contextlib import nullcontext as does_not_raise
+from importlib.abc import Loader
+from importlib.machinery import ModuleSpec
+from importlib.util import module_from_spec, spec_from_file_location
+import os
+from pathlib import Path
+import sys
+from typing import ContextManager, Dict, List
+
+import sqlalchemy
+
+import pytest
+from pytest import raises
+
+from ensembl.compara.filesys import file_cmp
+
+
+script_path = Path(__file__).parents[3] / "scripts" / "pipeline" / "orthology_benchmark.py"
+script_name = script_path.stem
+script_spec = spec_from_file_location(script_name, script_path)
+
+if not isinstance(script_spec, ModuleSpec):
+    raise ImportError(f"ModuleSpec not created for module file '{script_path}'")
+if not isinstance(script_spec.loader, Loader):
+    raise ImportError(f"no loader found for module file '{script_path}'")
+
+orthology_benchmark_module = module_from_spec(script_spec)
+sys.modules[script_name] = orthology_benchmark_module
+script_spec.loader.exec_module(orthology_benchmark_module)
+
+# pylint: disable=import-error,wrong-import-position
+
+import orthology_benchmark  # type: ignore
+
+# pylint: enable=import-error,wrong-import-position
+
+
+@pytest.mark.parametrize(
+    "multi_dbs",
+    [
+        [{'src': 'core/gallus_gallus_core_99_6', 'name': 'gallus_gallus_core_99_6'},
+         {'src': 'core/homo_sapiens_core_99_38', 'name': 'homo_sapiens_core_99_38'}]
+    ],
+    indirect=True
+)
+class TestDumpGenomes:
+    """Tests :func:`orthology_benchmark.dump_genomes()` function.
+
+    Attributes:
+        core_dbs: A set of test core databases.
+
+    """
+
+    core_dbs = {} # type: Dict
+
+    # autouse=True makes this fixture be executed before any test_* method of this class, and scope='class' to
+    # execute it only once per class parametrization
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, multi_dbs: Dict) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            multi_dbs: Dictionary of unit test databases (fixture).
+
+        """
+        type(self).core_dbs = multi_dbs
+
+    @pytest.mark.parametrize(
+        "species_list, species_set_name, expectation",
+        [
+            (["gallus_gallus", "homo_sapiens"], "default", does_not_raise()),
+            (["homo_sapiens", "zea_mays"], "default", raises(FileExistsError)),
+            (["felis_catus", "zea_mays"], "test", raises(RuntimeError,
+                    match=r"No cores found for the species set 'test' on the specified host."))
+        ]
+    )
+    def test_dump_genomes(self, species_list: List[str], species_set_name: str,
+                          tmp_dir: Path, expectation: ContextManager) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` function
+        when connection to the server host and port can be established.
```

Comment:
As mentioned by @JAlvarezJarreta, it is recommended to make the first line of a docstring self-contained. Here is one alternative wording to bring this one all on a single line:
```suggestion
        """Tests :func:`orthology_benchmark.dump_genomes()` when server connection can be established.
```